syntax = "proto3";

message CalcDiscountParameters {
    int32 product_id = 1;
    oneof optionalUserId {
        int32 user_id = 2;
    }
    oneof optionalForceDiscountDebug {
        int32 force_discount_debug = 3;
    }
}

message CalcDiscountResult {
    int32 discount = 1;
}

service Discount {
    rpc CalcDiscount(CalcDiscountParameters) returns (CalcDiscountResult);
}
